api/v1alpha1/cdapmaster_types.go

```diff
@@ -172,6 +172,10 @@ type CDAPScalableServiceSpec struct {
 	CDAPServiceSpec `json:",inline"`
 	// Replicas is number of replicas for the service.
 	Replicas *int32 `json:"replicas,omitempty"`
+	// Containers define any additional containers a service has
+	// This is a list of containers and can be left blank
+	// A typical use is to add sidecars for a deployment
+	Containers []*corev1.Container `json:"containers,omitempty"`
```

Comment:
Can we move this to the CDAPServiceSpec? That way we don't have to duplicate code in statefulset and deployment.

Reply:
Thanks for your review @dli357 I have now moved the Containers definitions to the CDAPServiceSpec struct. Please let me know if there is anything else you would like me to change. Thank you